sky/resources.py

```diff
@@ -225,6 +225,7 @@ def __init__(
         self._set_memory(memory)
         self._set_accelerators(accelerators, accelerator_args)
 
+        # TODO: move these out of init to prevent repeated calls.
```

Comment:
Yes! We have it here for make sure every `Resources` creation is validated, but this is can cause significant latency in our SkyPilot operations. We should optimize this in a future PR. Maybe file an issue for this?